### PR TITLE
Use promises

### PR DESCRIPTION
**Pull request by kevva**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,10 +35,17 @@
...
+    "pify": "^2.2.0",
+    "pinkie-promise": "^1.0.0",
     "ps-list": "^2.0.0"
   },
   "devDependencies": {
     "ava": "*",
     "xo": "*"
+  },
+  "xo": {
+    "ignore": [
```
Comment (SamVerschueren) [inline package.json]: [`ignores`](https://github.com/sindresorhus/xo#ignores) instead of `ignore`.

File: test.js [hunk 2]
```diff
@@ -1,7 +1,6 @@
 import test from 'ava';
 import fn from './';
 
-test(t => {
-	t.plan(1);
-	fn(err => t.ifError(err));
+test(async t => {
+	await fn();
 });
```
Comment (sindresorhus) [inline test.js]: `test(async t => await fn());`

**Comment by sindresorhus** [comment]
Thanks @kevva!

This module is such an obscure thing so I'll just go with minimum Node 4 and drop the ponyfill. Mind doing the changes?

**Comment by kevva** [comment]
@sindresorhus, yes, np.

File: index.js [hunk 3]
```diff
@@ -11,20 +12,10 @@ if (process.platform === 'win32') {
...
-
-	psList(function (err, list) {
-		if (err) {
-			cb(err);
-			return;
-		}
+module.exports = function (opts) {
+	opts = opts || {};
 
+	return pify(psList)().then(list => {
```
Comment (sindresorhus) [inline index.js]: Bump `ps-list` to use promise version.

File: index.js [hunk 4]
```diff
@@ -33,16 +24,16 @@ module.exports = function (opts, cb) {
...
 			}) &&
 			x.cmd.indexOf('--type=renderer') !== -1 &&
 			x.cmd.indexOf('--extension-process') === -1;
-		}).map(function (x) {
+		}).map(x => {
 			return x.pid;
 		});
 
-		fkill(pids, cb);
+		return pify(fkill)(pids);
```
Comment (sindresorhus) [inline index.js]: Bump `fkill` to use promise version.

File: win.js [hunk 5]
```diff
@@ -11,25 +11,25 @@ module.exports = function (opts, cb) {
 
 	cb = cb || function () {};
 
-	var cmd = 'wmic process where Caption=\'chrome.exe\' get CommandLine,ProcessId /format:csv';
+	const cmd = 'wmic process where Caption=\'chrome.exe\' get CommandLine,ProcessId /format:csv';
 
 	childProcess.exec(cmd, {
 		maxBuffer: TEN_MEBIBYTE
-	}, function (err, stdout) {
+	}, (err, stdout) => {
```
Comment (sindresorhus) [inline win.js]: `pify` this

File: win.js [hunk 6]
```diff
@@ -11,25 +11,25 @@ module.exports = function (opts, cb) {
...
 		if (err) {
 			cb(err);
 			return;
 		}
 
-		neatCsv(stdout.trim(), function (err, data) {
+		neatCsv(stdout.trim(), (err, data) => {
```
Comment (sindresorhus) [inline win.js]: `pify` this

File: cli.js [hunk 7]
```diff
@@ -1,22 +1,16 @@
 #!/usr/bin/env node
+/* eslint-disable no-mixed-spaces-and-tabs */
```
Comment (sindresorhus) [inline cli.js]: This can now be removed as it was fixed in ESLint.

**Pull request closed without merging**